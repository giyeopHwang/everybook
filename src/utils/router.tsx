import { createBrowserRouter } from 'react-router-dom';

import Front from '@/components/front/Front.component';
import DashboardLayout from '@/components/dashboard/dashboard-layout/DashboardLayout.compoent';

export const router = createBrowserRouter([
  {
    path: '/',
    element: <Front />,
  },
  {
    path: 'dashboard',
    element: <DashboardLayout />,
    children: [
      {
        path: '',
        element: <div>가계부</div>,
      },
      {
        path: 'report',
        element: <div>보고서</div>,
      },
      {
        path: 'budget',
        element: <div>예산 관리</div>,
      },
      {
        path: 'account',
        element: <div>분류 관리</div>,
      },
    ],
  },
]);
