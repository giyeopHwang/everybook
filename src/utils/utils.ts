import dayjs from 'dayjs';

export const displayAmount = (
  amount: number,
  showAmount: boolean = true,
  placeholder: string = '*********'
) => {
  return showAmount ? amount.toLocaleString() : placeholder;
};

export const formatDate = (date: Date = new Date()) => {
  const day = ['일', '월', '화', '수', '목', '금', '토'][dayjs(date).day()];
  return dayjs(date).format(`YYYY년 M월 D일 (${day})`);
};
