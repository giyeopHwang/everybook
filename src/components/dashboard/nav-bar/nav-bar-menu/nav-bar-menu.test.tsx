import { screen } from '@testing-library/react';
import { renderWithWrapper } from '@/test/test.utils';
import NavBarMenu from './nav-bar-menu.component';

describe('NavBarMenu', () => {
  it('renders NavBarMenu component', () => {
    const navBarMenu = renderWithWrapper(<NavBarMenu />);
    expect(navBarMenu).toMatchSnapshot();
  });

  it('includes four menu items', () => {
    const menuItems = ['가계부', '보고서', '예산 관리', '분류 관리'];
    renderWithWrapper(<NavBarMenu />);

    menuItems.forEach((menuItem) => {
      expect(screen.getByText(menuItem)).toBeInTheDocument();
    });
  });
});
