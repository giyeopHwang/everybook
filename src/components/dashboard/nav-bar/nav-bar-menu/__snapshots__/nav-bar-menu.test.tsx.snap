// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBarMenu > renders NavBarMenu component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="css-a5gwss MuiBox-root css-8n1nqt"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1svkyu9 css-etlun3-MuiTypography-root-MuiLink-root"
          href="/dashboard"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1l6op67-MuiSvgIcon-root"
            data-testid="EditIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            가계부
          </p>
        </a>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1svkyu9 css-etlun3-MuiTypography-root-MuiLink-root"
          href="/dashboard/report"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1l6op67-MuiSvgIcon-root"
            data-testid="BarChartIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M4 9h4v11H4zm12 4h4v7h-4zm-6-9h4v16h-4z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            보고서
          </p>
        </a>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1svkyu9 css-etlun3-MuiTypography-root-MuiLink-root"
          href="/dashboard/budget"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1l6op67-MuiSvgIcon-root"
            data-testid="PieChartIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 2v20c-5.07-.5-9-4.79-9-10s3.93-9.5 9-10zm2.03 0v8.99H22c-.47-4.74-4.24-8.52-8.97-8.99zm0 11.01V22c4.74-.47 8.5-4.25 8.97-8.99h-8.97z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            예산 관리
          </p>
        </a>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1svkyu9 css-etlun3-MuiTypography-root-MuiLink-root"
          href="/dashboard/account"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1l6op67-MuiSvgIcon-root"
            data-testid="CategoryIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m12 2-5.5 9h11z"
            />
            <circle
              cx="17.5"
              cy="17.5"
              r="4.5"
            />
            <path
              d="M3 13.5h8v8H3z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            분류 관리
          </p>
        </a>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="css-a5gwss MuiBox-root css-8n1nqt"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1svkyu9 css-etlun3-MuiTypography-root-MuiLink-root"
        href="/dashboard"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1l6op67-MuiSvgIcon-root"
          data-testid="EditIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          가계부
        </p>
      </a>
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1svkyu9 css-etlun3-MuiTypography-root-MuiLink-root"
        href="/dashboard/report"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1l6op67-MuiSvgIcon-root"
          data-testid="BarChartIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M4 9h4v11H4zm12 4h4v7h-4zm-6-9h4v16h-4z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          보고서
        </p>
      </a>
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1svkyu9 css-etlun3-MuiTypography-root-MuiLink-root"
        href="/dashboard/budget"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1l6op67-MuiSvgIcon-root"
          data-testid="PieChartIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 2v20c-5.07-.5-9-4.79-9-10s3.93-9.5 9-10zm2.03 0v8.99H22c-.47-4.74-4.24-8.52-8.97-8.99zm0 11.01V22c4.74-.47 8.5-4.25 8.97-8.99h-8.97z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          예산 관리
        </p>
      </a>
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1svkyu9 css-etlun3-MuiTypography-root-MuiLink-root"
        href="/dashboard/account"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1l6op67-MuiSvgIcon-root"
          data-testid="CategoryIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m12 2-5.5 9h11z"
          />
          <circle
            cx="17.5"
            cy="17.5"
            r="4.5"
          />
          <path
            d="M3 13.5h8v8H3z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          분류 관리
        </p>
      </a>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
